### PR TITLE
add minor Dockerfile improvements

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #    docker build -f examples/$name/Dockerfile -t prometheus/golang-example-$name .
 
 # Builder image, where we build the example.
-FROM golang:1 AS builder
+FROM golang:1.12 AS builder
```
Comment (tariq1890) [inline Dockerfile]: I don't think it's a good idea to pin to golang 1 as breaking changes could happen once golang 1.13 is out.

File: Dockerfile [hunk 2]
```diff
@@ -15,9 +15,9 @@ WORKDIR /go/src/github.com/prometheus/client_golang/examples/simple
 RUN CGO_ENABLED=0 GOOS=linux go build -a -tags netgo -ldflags '-w'
 
 # Final image.
-FROM prom/busybox
+FROM quay.io/prometheus/busybox:latest
```
Comment (tariq1890) [inline Dockerfile]: Looking at prometheus/busybox repo, quay seems to be current hosted registry for the busybox image.

File: Dockerfile (hunk 1 above)
Comment (simonpasquier) [reply Dockerfile]: at the same time, client_golang should always work with the latest stable version of Go.

File: Dockerfile (hunk 2 above)
Comment (simonpasquier) [reply Dockerfile]: correct

File: Dockerfile (hunk 1 above)
Comment (beorn7) [reply Dockerfile]: And the dockerfile is only for the examples. I don't expect any problems with any Go 1.x version, given the Go stability guarantees.

Comment (beorn7) [reply Dockerfile]: You could even turn the argument around: We deliberately want to build the examples with the latest Go version. If something doesn't work, that's a bug.

Comment (tariq1890) [reply Dockerfile]: Makes sense. I have fixed that.

**Comment by tariq1890** [comment]
Np :)

**Pull request merged**